src/provinces/province.cpp

```diff
@@ -1714,7 +1714,9 @@ std::vector<dcon::province_id> make_land_path(sys::state& state, dcon::province_
 
 				if(other_prov.id.index() < state.province_definitions.first_sea_province.index()) { // is land
 					if(has_access_to_province(state, nation_as, other_prov)) {
-						float danger_factor = has_safe_access_to_province(state, nation_as, other_prov) ? 1.f : 4.f;
+						/* This will work fine for most instances, except, possibly, for allied nations or enemy ones */
+						auto armies = state.world.province_get_army_location(other_prov);
+						float danger_factor = (armies.begin() == armies.end() || (*armies.begin()).get_army().get_controller_from_army_control() == nation_as) ? 1.f : 4.f;
```

Comment:
this will result in trying to avoid allied and neutral armies as well

Reply:
attrition